garage/src/hooks/useNFTs.ts

```diff
@@ -1,40 +1,62 @@
 import { useEffect, useState } from "react";
-import { useZDK } from "./useZDK";
-import { ZDK } from "@zoralabs/zdk";
+import { Chain, chain as chains } from "wagmi";
+import {
+  Network,
+  Alchemy,
+  NftTokenType,
+  Nft,
+  NftContract,
+  NftExcludeFilters,
+  GetNftsForOwnerOptions,
+} from "alchemy-sdk";
+import { chain, deployment } from "../env";
+import { useQuery } from "@tanstack/react-query";
 
-type TokensQuery = Awaited<ReturnType<ZDK["tokens"]>>;
+const wagmiChainToNetwork = (c: Chain): Network => {
+  switch (c) {
+    case chains.mainnet:
+      return Network.ETH_MAINNET;
+    case chains.goerli:
+      return Network.ETH_GOERLI;
+    case chains.polygon:
+      return Network.MATIC_MAINNET;
+    case chains.polygonMumbai:
+      return Network.MATIC_MUMBAI;
+    default:
+      throw new Error(`wagmiChainToNetwork unsupported chain, ${c}`);
+  }
+};
 
-type Unpacked<T> = T extends (infer U)[] ? U : T;
+const settings = {
+  apiKey: import.meta.env.VITE_ALCHEMY_ID,
+  network: wagmiChainToNetwork(chain),
+};
 
-type TokensQueryNode = Unpacked<TokensQuery["tokens"]["nodes"]>;
+export const alchemy = new Alchemy(settings);
 
 export interface NFT {
+  type: "ERC721" | "ERC1155" | "UNKNOWN";
   contract: string;
   tokenId: string;
   name?: string;
   imageUrl?: string;
+  imageSvgData?: string;
 }
 
-const queryNodeToNFT = (node: TokensQueryNode): NFT => {
-  const { tokenId, collectionAddress } = node.token;
-  const tokenName = node?.token.name;
-  const fallbackName = node
-    ? `${node?.token?.tokenContract?.name} #${node?.token?.tokenId}`
-    : undefined;
-  const media = node?.token.image?.mediaEncoding as { thumbnail: string };
-  const imageUrl = media?.thumbnail;
+const toNFT = (data: Nft): NFT => {
+  const { contract, tokenId, title, media } = data;
 
   return {
-    contract: collectionAddress,
+    type: contract.tokenType,
+    contract: contract.address,
     tokenId,
-    name: tokenName || fallbackName,
-    imageUrl,
+    name: title,
+    imageUrl: media.length ? media[0].gateway : undefined,
+    imageSvgData: data.rawMetadata?.svg_image_data,
```

Comment:
👍 . playing with something similar in animation_url, I noticed that one of my NFTs has a video for the `image` metadata (no real static image provided). i'm not sure how common that is... seems like they should have used `animation_url` for that, but looking into adding a video element to handle mp4 and webm files.